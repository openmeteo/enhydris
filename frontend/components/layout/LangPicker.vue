<template>
  <div class="navbar-item has-dropdown is-hoverable">
    <a class="navbar-link langpicker">{{ $t("language_picker") }} </a>
    <div class="navbar-dropdown is-hidden-mobile">
      <div>
        <nuxt-link
          v-if="currentLanguage != 'en'"
          class="navbar-item"
          :to="switchLocalePath('en')"
        >
          <img src="~/static/flags/us.svg" class="flagIcon" /> English
        </nuxt-link>
        <nuxt-link
          v-if="currentLanguage != 'el'"
          class="navbar-item"
          :to="switchLocalePath('el')"
        >
          <img src="~/static/flags/el.svg" class="flagIcon" /> Ελληνικά
        </nuxt-link>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "LangPicker",
  computed: {
    currentLanguage() {
      return this.$i18n.locale || "en";
    }
  }
};
</script>

<style scoped>
.langpicker {
  min-width: 120px;
}

.flagIcon {
  width: 20px;
  height: 20px;
  margin-right: 10px;
}
</style>
