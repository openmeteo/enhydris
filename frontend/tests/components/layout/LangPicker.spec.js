import { mount, RouterLinkStub } from "@vue/test-utils";
import LangPicker from "@/components/layout/LangPicker";

describe("LangPicker with locale en", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(LangPicker, {
      mocks: {
        $t: msg => msg,
        $i18n: { locale: "en" },
        switchLocalePath: msg => msg,
        localePath: msg => msg
      },
      stubs: {
        NuxtLink: RouterLinkStub
      }
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("contains link to Greek", () => {
    expect(wrapper.find(RouterLinkStub).props().to).toBe("el");
  });
});
