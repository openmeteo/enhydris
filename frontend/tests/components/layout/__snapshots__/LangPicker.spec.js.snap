// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangPicker with locale en renders correctly 1`] = `
<div
  class="navbar-item has-dropdown is-hoverable"
>
  <a
    class="navbar-link langpicker"
  >
    language_picker 
  </a>
   
  <div
    class="navbar-dropdown is-hidden-mobile"
  >
    <div>
      <!---->
       
      <a
        class="navbar-item"
      >
        <img
          class="flagIcon"
          src="~/static/flags/el.svg"
        />
         Ελληνικά
      
      </a>
    </div>
  </div>
</div>
`;
